tests/test_dashboard.py

```
@@ -0,0 +1,558 @@
+        with patch("asyncio.sleep", side_effect=WebSocketDisconnect()):
+            await dashboard_controller.handle_websocket(mock_ws)
+
+        mock_ws.accept.assert_called_once()
+        # Connection should be removed from set after disconnect
+        assert mock_ws not in dashboard_controller._websocket_connections
+
+    async def test_handle_websocket_streams_new_events(
+        self,
+        dashboard_controller: MetricsDashboardController,
+        mock_db_manager: Mock,
+        _mock_logger: Mock,
```

Ruff (0.14.5)

Fixture `_mock_logger` without value is injected as parameter, use `@pytest.mark.usefixtures` instead

(PT019)